### ISSUE TITLE
[RFC] Checkbox implementation

### ISSUE THREAD

**Issue by rfbezerra** [Rocketseat/unform#20]
Im thinking on the implementation of the checkbox functionality, and ended in this idea:
* work with an array of options, like select field
* options array will contain id and label (each option)
* getValue will return an array of checked ids

Well, let me know what you think about

**Comment by diego3g** [comment]
Should we use the same component for checkbox and radio? I mean, something like select that we can have multiple or not.

**Comment by rfbezerra** [comment]
I thought about it, and make some sense.
One concern about it is when I have a single option and want a checkbox, like an "Accept terms...". In that option, Ill have a component with only one option with a _multiple_ property to allow checkbox.
Obviously, it is fully customizable.

Ill publish an PR with it soon and than we discuss more

**Comment by rfbezerra** [comment]
PR #27 

Some rules:
- returns for multiple selection (checkboxes) are _undefined_ or an _array_
- returns for single selection (radios) are _undefined_ or an _string_
- if only one option is specified, checkbox will be used
- labels are optional

**Comment by diego3g** [comment]
I was thinking about this issue and the way unform works now. Basically unform only supports having one `ref` per form element, but what if we supported multiple refs per element and then the result comes in one array?

We could pass something like:

```js
registerField({ name: fieldName, ref: [ref1, ref2, ref3], path: "value" });
```

Then Unform could translate this value to: `[1, 2, 3]`.

What do you think?

**Comment by rfbezerra** [comment]
Well, in my opinion, the actual solution (only one ref) is better.
If we have to pass multiple refs, maybe we have to pass multiple path too.

Why not a getValue function like before or something like that?

**Comment by diego3g** [comment]
With the actual implementation with `registerField` inside useEffect `getValue` was not returning the actual value of the element so instead of using it, we changed to this ref approach.

Accepting multiple refs doesn't need multiple paths tho, since all the refs can have the same path to value.

I don't know, but with refs we have zero rerenders in React, i like this approach.

**Comment by diego3g** [comment]
What do you guys think about it? I mean by adding multiple refs to one form element, so checkbox component will look like this:
```js
registerField({ 
  name: fieldName, 
  ref: [ref1, ref2, ref3], 
  path: "value"
});
```
The main problem here is that every checkbox has a value, but they are not always selected, and we need to get the value only from selected ones, so `parseValue` couldn't do this job as he only receives the value itself, not the input reference.

Maybe `parseValue` could receive `ref` as a second parameter to search for selected property inside the checkbox?

---

Now, speaking about the implementation itself, maybe we could have a Choice (i don't like this name) component that is responsible for both checkbox and radios, so checkbox when multiple, radio when single, but maybe we could have and Boolean component that only renders a checkbox alone in case of "Accept terms" or "Change password".

Can you share your thoughts about this RFC @jpdemagalhaes and @pellizzetti?

**Comment by rfbezerra** [comment]
Hi, I tried to put your idea in place
Can you check this [this branch](https://github.com/rfbezerra/unform/tree/feature/Multiple_refs_on_register_field) to see if it fits your idea?

I added multiple refs to a field and an optional method to extract value from refs (getValues).
I think that the parseValues have another context (maybe concat values to backend, convert strings to numbers, etc...)

ps.: I dont like Choice name too, but I have no better name for now.

**Comment by diego3g** [comment]
@rfbezerra Great! Maybe you can send a PR and we can discuss more about the implementation there?

**Comment by Tabgyn** [comment]
About the name, how about we call Switch?

**Comment by jeancabral** [comment]
OMG! All that I need. When will this be released?

**Comment by jpedroschmitz** [comment]
@jeancabral, it was released yesterday, in v1.6.1  🎉  If you want to use, do not forget to upgrade to the correct version. I am also closing this issue due to the fact it was merged in #110.
